src/olympia/api/paginator.py

```diff
@@ -89,3 +89,23 @@ def get_paginated_response(self, data):
 class ESPageNumberPagination(CustomPageNumberPagination):
     """Custom pagination implementation to hook in our `ESPaginator`."""
     django_paginator_class = ESPaginator
+
+
+class OneOrZeroPageNumberPagination(CustomPageNumberPagination):
+    """Fake pagination that returns a result object like
+    CustomPageNumberPagination, but for special cases where we know we're never
+    going to have more than one object in the results anyway.
+    """
+    def paginate_queryset(self, queryset, request, view=None):
+        return list(queryset[:1])
```

Comment:
I honestly wonder if we should have an `assert` here that makes sure the queryset doesn't contain more than one value? I feel like failing silently is quite bad if the used view is only supposed to return one value.